src/commands/cmd_dispatcher.c

```diff
@@ -57,18 +57,14 @@ int CommandDispatch(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 											REDISMODULE_CTX_FLAGS_LOADING));
 	if(execute_on_main_thread) {
 		// Run query on Redis main thread.
-		context = CommandCtx_New(ctx, NULL, gc, argv[2], argv, argc, is_replicated);
+		context = CommandCtx_New(ctx, NULL, command_name, gc, argv[2], argv, argc, is_replicated);
 		handler(context);
 	} else {
 		// Run query on a dedicated thread.
 		RedisModuleBlockedClient *bc = RedisModule_BlockClient(ctx, NULL, NULL, NULL, 0);
-		context = CommandCtx_New(NULL, bc, gc, argv[2], argv, argc, is_replicated);
+		context = CommandCtx_New(NULL, bc, command_name, gc, argv[2], argv, argc, is_replicated);
 		thpool_add_work(_thpool, handler, context);
 	}
 
-	// Replicate the command to slaves and AOF.
-	// If the query is read-only, slaves will do nothing after parsing.
-	// TODO: only replicate if query is a writer.
-	if(cmd == CMD_QUERY || cmd == CMD_PROFILE) RedisModule_ReplicateVerbatim(ctx);
```

Comment:
This is perfect, now we're able to not replicate in-case of a read query, please confirm this is the case.
In addition please make sure the logic for dropping execution of a replicated read command from a slave is removed.

Reply:
@DvirDukhan, see cmd_query.c 
```
bool readonly = AST_ReadOnly(parse_result);
// If we are a replica and the query is read-only, no work needs to be done.
if(readonly && qctx->replicated_command) goto cleanup;
```
this handles the case were a slave process a read query sent to it from a master, which now shouldn't happen.